crates/ruff/src/printer.rs

```diff
@@ -389,22 +389,30 @@ impl Printer {
             }
 
             let context = EmitterContext::new(&diagnostics.notebook_indexes);
-            let format = if preview {
-                DiagnosticFormat::Full
+            if preview {
+                let config = DisplayDiagnosticConfig::default()
+                    .preview(preview)
+                    .hide_severity(true)
+                    .color(!cfg!(test) && colored::control::SHOULD_COLORIZE.should_colorize())
+                    .with_show_fix_status(show_fix_status(self.fix_mode, fixables.as_ref()))
+                    .with_fix_applicability(self.unsafe_fixes.required_applicability())
+                    .show_fix_diff(preview);
+
+                render_diagnostics(writer, self.format, config, &context, &diagnostics.inner)?;
             } else {
-                DiagnosticFormat::Concise
-            };
-            let config = DisplayDiagnosticConfig::default()
-                .hide_severity(true)
-                .color(!cfg!(test) && colored::control::SHOULD_COLORIZE.should_colorize())
-                .with_show_fix_status(show_fix_status(self.fix_mode, fixables.as_ref()))
-                .format(format)
-                .with_fix_applicability(self.unsafe_fixes.required_applicability());
-            write!(
-                writer,
-                "{}",
-                DisplayDiagnostics::new(&context, &config, &diagnostics.inner)
-            )?;
+                let config = DisplayDiagnosticConfig::default()
+                    .hide_severity(true)
+                    .color(!cfg!(test) && colored::control::SHOULD_COLORIZE.should_colorize())
+                    .with_show_fix_status(show_fix_status(self.fix_mode, fixables.as_ref()))
+                    .format(DiagnosticFormat::Concise)
+                    .with_fix_applicability(self.unsafe_fixes.required_applicability());
+
+                write!(
+                    writer,
+                    "{}",
+                    DisplayDiagnostics::new(&context, &config, &diagnostics.inner)
+                )?;
```

Comment:
How's `DisplayDiagnostics` different from `render_diagnostics`? Or why can't we do

```
let format = if preview {
	self.format
} else {
	DiagnosticFormat::Concise
};

let config = DisplayDiagnosticConfig::default()
    .preview(preview)
    .hide_severity(true)
    .color(!cfg!(test) && colored::control::SHOULD_COLORIZE.should_colorize())
    .with_show_fix_status(show_fix_status(self.fix_mode, fixables.as_ref()))
    .with_fix_applicability(self.unsafe_fixes.required_applicability())
    .show_fix_diff(preview);

render_diagnostics(writer, self.format, config, &context, &diagnostics.inner)?;
```